### PR TITLE
refactor: Crawler/sourcerepos

### PR DESCRIPTION
**Pull request by natea123**
closes [#212](https://github.com/google/gcp_scanner/issues/212)
ref to [#153](https://github.com/google/gcp_scanner/issues/153)

### CODE REVIEW

**Review by ZetaTwo — APPROVED** [review]

**Comment by ZetaTwo** [comment]
Tests are failing. Please check.

File: src/gcp_scanner/crawl.py [hunk 1]
```diff
@@ -703,36 +703,37 @@ def list_services(project_id: str, service: discovery.Resource) -> List[Any]:
   return list_of_services
 
 
-def list_sourcerepo(project_id: str,
-                    service: discovery.Resource) -> List[Any]:
-  """Retrieve a list of cloud source repositories enabled in the project.
 
+def list_dns_policies(project_id: str,
```
Comment (ZetaTwo) [inline src/gcp_scanner/crawl.py:707]: The rest of the change is about source repos but this lists dns policies? Did you mix up the refactorings?

Comment (natea123) [reply src/gcp_scanner/crawl.py:707]: Added the dns_policies to remove the merge conflict caused by #214

Comment (natea123) [reply src/gcp_scanner/crawl.py]: Apologies, I didn't realize the merge conflict was caused by #214 - updated crawl.py to match main

File: src/gcp_scanner/crawl.py [hunk 2]
```diff
@@ -700,39 +700,4 @@ def list_services(project_id: str, service: discovery.Resource) -> List[Any]:
...
-      request = service.projects().repos().list_next(
-        previous_request=request,
-        previous_response=response
-      )
-  except Exception:
-    logging.info("Failed to retrieve source repos for project %s", project_id)
-    logging.info(sys.exc_info())
-
-  return list_of_repos
+  return list_of_services
```
Comment (ZetaTwo) [inline src/gcp_scanner/crawl.py:703]: Add a newline

**Review by ZetaTwo — CHANGES_REQUESTED** [review]
Looks good. small nit.

File: src/gcp_scanner/crawl.py (hunk 2 above)
Comment (natea123) [reply src/gcp_scanner/crawl.py]: ack

**Review by ZetaTwo — APPROVED** [review]

**Pull request merged**